stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/extensions/LiveDataExtensions.kt

```diff
@@ -0,0 +1,10 @@
+package io.getstream.chat.android.livedata.extensions
+
+import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+internal suspend fun <T : Any?> MutableLiveData<T>.setOnUi(valueProvider: () -> T) =
+    withContext(Dispatchers.Main) {
+        value = valueProvider()
+    }
```

Comment:
Is there a specific reason why we want the code inside `valueProvider` to be executed on the UI thread too? Seems like it'd be better to generate the value on a background thread if possible, and do just the setting on the UI thread.

Reply:
Ah so you're waiting in the IO thread to write on the UI/Main thread. Well that solves the problem :)
Feels a bit hacky, but it works. 

A hard coded dispatcher is discouraged for testing purposes right?

Reply:
Hardcoded dispatcher can be real problem in testing. It's not ideal solve for the problem. When we will have switched to flow it will be much easier to remove

Reply:
I think Marton is right that lambda can be heavy for UI. I changed it

Reply:
We'll likely always swap the Main dispatcher in tests with `Dispatchers.setMain`, essentially in a static way, so it's _fine_ to use it directly. For every other dispatcher, they should really be injected so we can swap them out in tests.

We could even go as far as having a standardized `DispatcherProvider` interface that holds all the ones we use, and inject that everywhere. Then it would be easy to have a fake implementation of that for testing that we swap in.

---

Let's not forget the original comment up here though, that passing in the value `T` instead of the lambda that generates it would be a good idea here, so that the code in the lambda is not executed on the main thread 😄

Reply:
I like idea of DispatcherProvider. It gives a good level of flexibility

Reply:
Now that I think of it, proper injecting would be difficult in some places (e.g. individual functions that use Dispatchers, or small utility classes like `Call` implementations). I guess we can go the same way as the coroutine library works, have a static instance like `DispatcherProvider.instance().Main`, and then swap the Dispatchers inside inside there for tests.